src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -271,24 +269,27 @@ def sync_deploy_root_with_stage(self, role: str) -> DiffResult:
             )
 
         # Perform a diff operation and display results to the user for informational purposes
-        log.info(
-            'Performing a diff between the Snowflake stage and your local deploy_root ("%s") directory.',
-            self.deploy_root,
+        cc.step(
+            "Performing a diff between the Snowflake stage and your local deploy_root ('%s') directory."
+            % self.deploy_root
         )
         diff: DiffResult = stage_diff(self.deploy_root, self.stage_fqn)
-        log.info("Listing results of diff:")
-        log.info("New files only on your local: %s", ",".join(diff.only_local))
-        log.info("New files only on the stage: %s", ",".join(diff.only_on_stage))
-        log.info(
-            "Existing files modified or status unknown: %s", ",".join(diff.different)
-        )
-        log.info("Existing files identical to the stage: %s", ",".join(diff.identical))
+        with cc.phase("Listing results of diff:"):
+            cc.step("New files only on your local: %s" % ",".join(diff.only_local))
+            cc.step("New files only on the stage: %s" % ",".join(diff.only_on_stage))
+            cc.step(
+                "Existing files modified or status unknown: %s"
+                % ",".join(diff.different),
+            )
```

Comment:
I know that will make a code a little odd, but IMHO that's a single step, i.e. nothing should be interleaved between each line. I'd prefer to assemble the output once, then print it as a unit.